descriptor/fingerprint/src/main/java/org/openscience/cdk/fingerprint/AtomPairs2DFingerprinter.java

```diff
@@ -0,0 +1,208 @@
+ * http://www.yapcwsoft.com/dd/padeldescriptor/
+ * 
+ */
+
+
+package org.openscience.cdk.fingerprint;
+
+import org.openscience.cdk.config.Elements;
+import org.openscience.cdk.exception.CDKException;
+import org.openscience.cdk.graph.AllPairsShortestPaths;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IAtomContainer;
+
+import java.util.ArrayList;
+import java.util.BitSet;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+//import org.openscience.cdk.graph.matrix.TopologicalMatrix;
+
+/**
+ * Generates an atom pair 2D fingerprint as implemented in PaDEL given an  {@link IAtomContainer}, that
+ * extends the {@link Fingerprinter}.
+ *
+ * @author Lyle Burgoon
+ * @cdk.created 2018-02-05
+ * @cdk.keyword fingerprint
+ * @cdk.keyword similarity
+ * @cdk.module fingerprint
+ * @cdk.githash
+ * @see org.openscience.cdk.fingerprint.Fingerprinter
+ */
+public class AtomPairs2DFingerprinter extends AbstractFingerprinter implements IFingerprinter {
+
+    private static final int      MAX_DISTANCE = 10;
+    private static final String[] atypes       = {"C", "N", "O", "S", "P", "F", "Cl", "Br", "I", "B", "Si", "X"};
+
+    private final Map<String, Integer> pathToBit = new HashMap<>();
+    private final Map<Integer, String> bitToPath = new HashMap<>();
+
+    public AtomPairs2DFingerprinter() {
+        for (int dist = 1; dist <= MAX_DISTANCE; dist++) {
+            for (int i = 0; i < atypes.length; i++) {
+                for (int j = i; j < atypes.length; j++) {
+                    final String key_name = dist + "_" + atypes[i] + "_" + atypes[j];
+                    pathToBit.put(key_name, pathToBit.size());
+                    bitToPath.put(bitToPath.size(), key_name);
+                }
+            }
+        }
+    }
+
+    @Override
+    public int getSize() {
+        return pathToBit.size();
+    }
+
+    private static boolean isHalogen(final IAtom atom) {
+        switch (atom.getAtomicNumber()) {
+            case 9:  // F
+            case 17: // Cl
+            case 35: // Br
+            case 53: // I
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    private static boolean include(final IAtom atom) {
+        switch (atom.getAtomicNumber()) {
+            case 5:  // B
+            case 6:  // C
+            case 7:  // N
+            case 8:  // O
+            case 14: // Si
+            case 15: // P
+            case 16: // S
+            case 9:  // F
+            case 17: // Cl
+            case 35: // Br
+            case 53: // I
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    private static String encodePath(int dist, IAtom a, IAtom b) {
+        return dist + "_" + a.getSymbol() + "_" + b.getSymbol();
+    }
+
+    private static String encodeHalPath(int dist, IAtom a, IAtom b) {
```

Comment:
I generally like to have some brief documentation on private methods too, to support maintenance of the code...